includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -335,8 +331,10 @@ public function get_upe_enabled_payment_method_ids() {
 	 * @return string[]
 	 */
 	public function get_upe_enabled_at_checkout_payment_method_ids( $order_id = null ) {
-		$available_method_ids = [];
-		foreach ( $this->get_upe_enabled_payment_method_ids() as $payment_method_id ) {
+		$capture                 = empty( $this->get_option( 'capture' ) ) || $this->get_option( 'capture' ) === 'yes';
+		$enabled_payment_methods = $capture ? $this->get_upe_enabled_payment_method_ids() : [ 'card' ];
```

Comment:
I love the idea of having an attribute on the class!

I'd like to mention - a merchant _could_ disable the `card` payment method.
So in this case, if they had the `card` payment method disabled and `capture` set to `'yes'`, they would get `card` at checkout.

Reply:
> So in this case, if they had the `card` payment method disabled and `capture` set to `'yes'`, they would get card at checkout.

That is how I interpreted the code here, yeah.

Is that the expected behavior in this case though? That behavior seems a bit counter-intuitive for merchants.

If I chose to not allow card payments on my store and only have 1 or more non-card UPE methods enabled which don't support manual capture, I wouldn't expect toggling the `capture` setting on would suddenly and silently disable my chosen UPE methods and enable `card` (which I explicitly turned off!) as the only payment option. I actually wouldn't expect to be able to toggle the `capture` setting at all in such a case. :thinking:

Reply:
>In the future, perhaps we could have a property in the UPE method classes for "supports manual capture" sort of like we have isReusable currently. 

Created #1956.

>If I chose to not allow card payments on my store and only have 1 or more non-card UPE methods enabled which don't support manual capture, I wouldn't expect toggling the capture setting on would suddenly and silently disable my chosen UPE methods and enable card (which I explicitly turned off!) as the only payment option. I actually wouldn't expect to be able to toggle the capture setting at all in such a case.

Agreed. We're discussing the scope of the toggling the capture settings behavior on p1631843780047400-slack-CHG7MTCAF I can draft a P2 to gather all these requirements and include this to make sure it doesn't get lost when a decision is made.

Reply:
(honest question) for the time being, should we just try and honor what the merchant selected in their settings?

They might get an error at checkout, but at least it would be because of their selections.
I don't think otherwise they'd be happy to get the "card" payment method at checkout, even when they explicitly disabled it.

Reply:
>We could do so by filtering (in this class method) all the payment methods that are not card when $capture is truthy. That way, if card is disabled, the whole array would also be empty.

Good point, thanks for the suggestion! However, I think Stripe doesn't let us do that. I enabled manual capture and disabled card (having other payment methods enabled). With the implementation below, Stripe automatically fallbacks to `['card']`.
```php
$manual_capture_methods  = in_array('card', $this->get_upe_enabled_payment_method_ids()) ? ['card'] : [];
$enabled_payment_methods = $capture ? $this->get_upe_enabled_payment_method_ids() : $manual_capture_methods;```

Reply:
Ah, gotcha - thanks for checking that!

Reply:
I added [a note about this](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-testing-instructions-for-the-WooCommerce-Stripe-payment-gateway-5.6.0#merchant) in the GlobalStep testing docs.